cmd/restic/find.go

```diff
@@ -14,17 +15,31 @@ func initMultiSnapshotFilter(flags *pflag.FlagSet, filt *restic.SnapshotFilter,
 	if !addHostShorthand {
 		hostShorthand = ""
 	}
-	flags.StringArrayVarP(&filt.Hosts, "host", hostShorthand, nil, "only consider snapshots for this `host` (can be specified multiple times)")
+	flags.StringArrayVarP(&filt.Hosts, "host", hostShorthand, nil, "only consider snapshots for this `host` (can be specified multiple times) (default: $RESTIC_HOST)")
 	flags.Var(&filt.Tags, "tag", "only consider snapshots including `tag[,tag,...]` (can be specified multiple times)")
 	flags.StringArrayVar(&filt.Paths, "path", nil, "only consider snapshots including this (absolute) `path` (can be specified multiple times)")
+
+	if len(filt.Hosts) == 0 {
```

Comment:
Wrapping the following code in the if is unnecessary. The cli flag configuration and the cli parsing are two separate steps. At this point, `filt.Hosts` is always empty.

It does still work, as cobra internally tracks whether it has already applied an option or whether `filt.Hosts` still has its default value. So, just remove the `if len(filt.Hosts) == 0 {` check. The same also applies for `initSingleSnapshotFilter`.